opm/grid/common/ZoltanPartition.cpp

```diff
@@ -307,20 +307,21 @@ zoltanGraphPartitionGridOnRoot(const CpGrid& cpgrid,
                                bool allowDistributedWells,
                                const std::map<std::string,std::string>& params)
 {
-    int rc = ZOLTAN_OK - 1;
     float ver = 0;
     struct Zoltan_Struct *zz;
     int changes, numGidEntries, numLidEntries, numImport, numExport;
     ZOLTAN_ID_PTR importGlobalGids, importLocalGids, exportGlobalGids, exportLocalGids;
     int *importProcs, *importToPart, *exportProcs, *exportToPart;
     int argc=0;
     char** argv = 0 ;
-    rc = Zoltan_Initialize(argc, argv, &ver);
-    zz = Zoltan_Create(cc);
-    if ( rc != ZOLTAN_OK )
-    {
+    int rc = Zoltan_Initialize(argc, argv, &ver);
+    if (rc != ZOLTAN_OK) {
         OPM_THROW(std::runtime_error, "Could not initialize Zoltan!");
     }
+    zz = Zoltan_Create(cc);
+    if (rc != ZOLTAN_OK) {
+        OPM_THROW(std::runtime_error, "Could not create Zoltan data structures!");
+    }
```

Comment:
This `rc` is the one from `Zoltan_Initialize()` above and won't be anything other than `ZOLTAN_OK` if we get here.  Did you mean `zz == nullptr` instead?